dts/bindings/misc/zephyr,morse-code.yaml

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2024 Freedom Veiculos Eletricos
+# Copyright (c) 2024 OS Systems
+# SPDX-License-Identifier: Apache-2.0
+
+description: Morse Code
+
+compatible: "zephyr,morse-code"
+
+include:
+  - name: base.yaml
+
+properties:
+  speed:
+    type: int
+    default: 20
+    description: Define the number of Words Per Minute (wps) should be send
+      per minute. This is defined using the standard word "PARIS". In the
+      default case means that "20 wpm" send the word "PARIS" (or precisely
+      "PARIS ") is sent 20 times in a minute.
+
+      In morse the "PARIS " is 50 units long and represented by
+      ".--. .- .-. .. .../" were
+
+        10 dits (.)              10 units
+        4 dahs (-)               12 units
+        9 intra-character spaces  9 units
+        4 inter-character spaces 12 units
+        1 word space (/)          7 units
+
+      In this case, the driver defines the time unit as 60/(50 * Speed). With a
+      default speed of 20 wps the time unit is equal to 0.06s.
+
+  gpios:
+    type: phandle-array
+    required: true
+    description: The GPIO configuration which will be used to send the morse code.
+
+  timer-unit:
+    type: phandle
+    required: true
+    description: The Counter Timer compatible that will be used to serialize
+      the data into morse code at "tone-out" output.
```

Comment:
I think everything above this is generic enough to be in its own .yml

This part is only one possible device type (i.e. it is bit-banged) and should probably be in a separate .yml that includes the generic part.